### PR TITLE
Rewrite platform specific dependencies in Cargo.toml

### PR DESCRIPTION
**Pull request by viccie30**
This PR adds support for rewriting [platform specific dependencies](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#platform-specific-dependencies).

I have added a test by copying `lib_with_path_dep_sdist` and changing `Cargo.toml`.

### CODE REVIEW

File: tests/run.rs [hunk 1]
```diff
@@ -479,6 +479,26 @@ fn lib_with_path_dep_sdist() {
...
+            "sdist_with_target_path_dep-0.1.0/local_dependencies/some_path_dep/src/lib.rs",
+            "sdist_with_target_path_dep-0.1.0/local_dependencies/transitive_path_dep/Cargo.toml",
+            "sdist_with_target_path_dep-0.1.0/local_dependencies/transitive_path_dep/src/lib.rs",
+            "sdist_with_target_path_dep-0.1.0/Cargo.toml",
+            "sdist_with_target_path_dep-0.1.0/Cargo.lock",
+            "sdist_with_target_path_dep-0.1.0/pyproject.toml",
+            "sdist_with_target_path_dep-0.1.0/src/lib.rs",
+            "sdist_with_target_path_dep-0.1.0/PKG-INFO",
+        ],
+        None,
```
Comment (messense) [inline tests/run.rs:497]: Can you add an assertion for rewritten `Cargo.toml` content?

Comment (viccie30) [reply tests/run.rs]: Of course: https://github.com/PyO3/maturin/blob/110625e66d81bf2cf23d39b065a1faf2e5a1730a/tests/run.rs#L486-L500

**Review by messense — APPROVED** [review]
Thanks!

bors r+

**Comment by messense** [comment]
bors retry

**Pull request merged**